scale/scheduler/scheduling/manager.py

```diff
@@ -336,94 +336,122 @@ def _process_queue(self, nodes, job_types, job_type_limits, job_type_resources,
         """
 
         scheduled_job_executions = []
-        ignore_job_type_ids = self._calculate_job_types_to_ignore(job_types, job_type_limits)
         started = now()
+        type_warnings = {}
+
+        # We can schedule as long as there are nodes
+        if nodes:
```

Comment:
I'm not a big fan of nesting the entire scheduler loop within this if block. 2 reasons:
1. It's really hard to review what actually changed now that every line is _changed_ due to whitspace.
2. Having the shortcut cases up front where the loop no-ops is nice for tracing logically in my head.